tests/integration/Statistics/Track/WooCommerceOrdersTest.php

```diff
@@ -0,0 +1,216 @@
+
+    expect(count(StatisticsWooCommerceOrders::findMany()))->equals(2);
+  }
+
+  function testItTracksPaymentOnlyForLatestClick() {
+    $latest_click = $this->createClick($this->link, $this->subscriber, 1);
+    $this->createClick($this->link, $this->subscriber, 3);
+    $this->createClick($this->link, $this->subscriber, 5);
+    $order_mock = $this->createOrderMock($this->subscriber->email);
+    $woocommerce_orders = new WooCommerceOrders($this->createWooCommerceHelperMock($order_mock));
+    $woocommerce_orders->trackPaidOrder($order_mock->get_id());
+
+    $order_stats = StatisticsWooCommerceOrders::findMany();
+    expect(count($order_stats))->equals(1);
+    expect($order_stats[0]->click_id)->equals($latest_click->id);
+  }
+
+  function testItTracksPaymentOnlyOnce() {
+    $this->createClick($this->link, $this->subscriber);
+    $order_mock = $this->createOrderMock($this->subscriber->email);
+    $woocommerce_orders = new WooCommerceOrders($this->createWooCommerceHelperMock($order_mock));
+    $woocommerce_orders->trackPaidOrder($order_mock->get_id());
+    $woocommerce_orders->trackPaidOrder($order_mock->get_id());
+    expect(count(StatisticsWooCommerceOrders::findMany()))->equals(1);
+  }
+
+  function testItDoesNotTrackPaymentWhenClickTooOld() {
+    $this->createClick($this->link, $this->subscriber, 20);
+    $order_mock = $this->createOrderMock($this->subscriber->email);
+    $woocommerce_orders = new WooCommerceOrders($this->createWooCommerceHelperMock($order_mock));
+    $woocommerce_orders->trackPaidOrder($order_mock->get_id());
+    expect(count(StatisticsWooCommerceOrders::findMany()))->equals(0);
+  }
+
+  function testItDoesNotTrackPaymentForDifferentEmail() {
+    $this->createClick($this->link, $this->subscriber);
+    $order_mock = $this->createOrderMock('different.email@example.com');
+    $woocommerce_orders = new WooCommerceOrders($this->createWooCommerceHelperMock($order_mock));
+    $woocommerce_orders->trackPaidOrder($order_mock->get_id());
+    expect(count(StatisticsWooCommerceOrders::findMany()))->equals(0);
+  }
+
+  function testItDoesNotTrackPaymentWithZeroCost() {
+    $this->createClick($this->link, $this->subscriber);
+    $order_mock = $this->createOrderMock($this->subscriber->email, 0);
+    $woocommerce_orders = new WooCommerceOrders($this->createWooCommerceHelperMock($order_mock));
+    $woocommerce_orders->trackPaidOrder($order_mock->get_id());
+    expect(count(StatisticsWooCommerceOrders::findMany()))->equals(0);
+  }
+
+  function testItDoesNotTrackPaymentWhenClickNewerThanOrder() {
+    $this->createClick($this->link, $this->subscriber, 0);
+    $order_mock = $this->createOrderMock($this->subscriber->email);
+    $woocommerce_orders = new WooCommerceOrders($this->createWooCommerceHelperMock($order_mock));
+    $woocommerce_orders->trackPaidOrder($order_mock->get_id());
+    expect(count(StatisticsWooCommerceOrders::findMany()))->equals(0);
+  }
+
+  function _after() {
```

Comment:
Shouldn't we clean those tables before the test? If your test requires some tables to be empty it should ensure it itself. If something else doesn't clean, this test will fail even though there is nothing wrong with it. This causes flakiness.

Reply:
The test only cleans data it created. This is the same case for most of other tests. We don't clean tables before normally.

Reply:
I know we don't do that normally. But I am saying we should start doing that.

Reply:
I'm not sure if we should start doing this here, maybe it would be better to leave this for a separate discussion. My opinion is that it should be resolved systematically in a way that every test would start with an empty DB, create what it needs and there would be no before/after cleaning at all (which also makes the code shorter).

Reply:
We wouldn't start it here, a few other tests like `\MailPoet\Test\API\JSON\v1\SubscribersTest` are already cleaning in `_before`

> it should be resolved systematically in a way that every test would start with an empty DB

it could be resolved systematically, but until it is (or if it never is) done we can have the cleanup done in the better place. 
And if we decide to remove the code, it doesn't matter if it was in before or after.

Reply:
Alright 👍